### PR TITLE
feat(Transfer): add new option for location field

### PR DESCRIPTION
**Pull request by stonebuzz**
Add new location option to transfer

image

* Empty the location
* Preserve (default behavior and same as before)

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !29010

### CODE REVIEW

File: tests/functional/Transfer.php [hunk 1]
```diff
@@ -539,4 +539,97 @@ public function testKeepCertificateOption(
...
+        $this->boolean($ticket->getFromDB($ticket_id))->isTrue();
+
+
+        //transer to another entity
+        $transfer = new \Transfer();
+        $transfer->getFromDB(1);
+
+        //update transfer model to keep location
+        $transfer->fields["keep_location"] = 1;
+        $transfer->update($transfer->fields);
```
Comment (cedric-anne) [inline tests/functional/Transfer.php:580]:
```suggestion
        $this->boolean($transfer->update($transfer->fields))->isTrue();
```

File: tests/functional/Transfer.php [hunk 2]
```diff
@@ -539,4 +539,97 @@ public function testKeepCertificateOption(
...
+            'locations_id' => $location_id,
+            'entities_id'  => $fentity
+        ]);
+        $this->integer($ticket_id)->isGreaterThan(0);
+        $this->boolean($ticket->getFromDB($ticket_id))->isTrue();
+
+
+        //transer to another entity
+        $transfer = new \Transfer();
+        $transfer->getFromDB(1);
```
Comment (cedric-anne) [inline tests/functional/Transfer.php:622]:
```suggestion
        $this->boolean($transfer->getFromDB(1))->isTrue();
```

File: tests/functional/Transfer.php [hunk 3]
```diff
@@ -539,4 +539,97 @@ public function testKeepCertificateOption(
...
+
+        //update transfer model to empty location
+        $transfer->fields["keep_location"] = 0;
+        $transfer->update($transfer->fields);
```
Comment (cedric-anne) [inline tests/functional/Transfer.php:626]:
```suggestion
        $this->boolean($transfer->update($transfer->fields))->isTrue();
```

File: install/migrations/update_10.0.10_to_10.0.11/transfer.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+
+
+$migration->addField('glpi_transfers', 'keep_location', "int", ['value' => '1']);
```
[the remark this answers is not in the document]
Comment (trasher) [reply install/migrations/update_10.0.10_to_10.0.11/transfer.php:42]: I do not think it's about other fields default, but about what we want to do with that feature... Keeping location as default makes sense.

[the remark this answers is not in the document]
Comment (trasher) [reply install/migrations/update_10.0.10_to_10.0.11/transfer.php:42]: If you really want to use '0' always per default, it should be documented (as a comment) somewhere, and therefore the option name would become like `empty_location` and then defaults to `0`.

[the remark this answers is not in the document]
Comment (cedric-anne) [reply install/migrations/update_10.0.10_to_10.0.11/transfer.php:42]: OK, so `keep_location` with default value to 1 is OK.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**